src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -67,11 +82,21 @@ StartRemoteTransactionBegin(struct MultiConnection *connection)
 	 */
 	distributedTransactionId = GetCurrentDistributedTransactionId();
 	appendStringInfo(beginAndSetDistributedTransactionId,
-					 "SELECT assign_distributed_transaction_id(%d, %ld, '%s')",
+					 "SELECT assign_distributed_transaction_id(%d, %ld, '%s');",
 					 distributedTransactionId->initiatorNodeIdentifier,
 					 distributedTransactionId->transactionNumber,
 					 timestamptz_to_str(distributedTransactionId->timestamp));
 
+	/* append in-progress savepoints for this transaction */
+	activeSubXacts = ActiveSubXacts();
+	transaction->lastSuccessfulSubXact = TopSubTransactionId;
+	foreach(subIdCell, activeSubXacts)
+	{
+		SubTransactionId subId = lfirst_int(subIdCell);
+		appendStringInfo(beginAndSetDistributedTransactionId,
+						 "SAVEPOINT savepoint_%u;", subId);
+		transaction->lastSuccessfulSubXact = subId;
```

Comment:
Done.
(I was assuming if the whole beginAndSetDistributedTransactionId isn't successful then we don't need lastSuccessfulSubXact at all. Am I missing something?)